src/selectors/right-clicked-call-node.js

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import { createSelector } from 'reselect';
+import { getCallNodeIndexFromPath } from '../profile-logic/profile-data';
+import { getThreadSelectors } from './per-thread';
+import { getProfileViewOptions } from './profile';
+
+import type { Thread, ThreadIndex } from '../types/profile';
+import type {
+  IndexIntoCallNodeTable,
+  CallNodePath,
+  CallNodeInfo,
+} from '../types/profile-derived';
+import type { Selector, DangerousSelectorWithArguments } from '../types/store';
+
+export type RightClickedCallNodeInfo = {|
+  +threadIndex: ThreadIndex,
+  +thread: Thread,
+  +callNodePath: CallNodePath,
+  +callNodeIndex: IndexIntoCallNodeTable,
+  +callNodeInfo: CallNodeInfo,
+|};
+
+export const getRightClickedCallNodeInfo: Selector<RightClickedCallNodeInfo | null> = createSelector(
+  state => state,
```

Comment:
so, I know I suggested this in my previous comment. Now I'm not so sure about that anymore.
The problem is that this breaks memoization: memoization will only work if `state` is the same.

What problem does that give? Mainly that all the code will be run more than necessary.
Is that a big problem? Depends on the code :-)

We can decompose it:
* getProfileViewOptions => it's memoized elsewhere
* thread selectors, getFilteredThread, getCallNodeInfo => memoized elsewhere
* getCallNodeIndexFromPath => not memoized
* return object => not memoized

The fact that the return object isn't memoized will trigger needless rerenders on the context menu component... except in our specific case we return null very quickly most of the time.

I need to look at it closer before giving a more helpful advice.
One obvious question is: should we move these selectors to selectedThreadSelectors after all? (while keeping the state where you've put it, which was the main problem in the previous code). Then it would go to stack-sample.js for thi one, and markers.js for the marker selector.

Does that matter anyway, given this returns null most of the time?

Would you be open to try this change and see if this works?

Reply:
Yeah, sure, I'll try to do that! After thinking about moving this stuff to thread selectors I think it wouldn't be as tricky to refactors the components afterwards as I thiught initially as not that many of them actually need anything but indexes that are already available in the right-clicked reducer 👍